operate/common/src/main/java/io/camunda/operate/conditions/DatabaseCondition.java

```diff
@@ -14,7 +14,7 @@
 
 public abstract class DatabaseCondition implements Condition {
 
-  public static final String DATABASE_PROPERTY = "camunda.operate.database";
+  public static final String DATABASE_PROPERTY = "camunda.data.secondary-storage.type";
```

Comment:
:x: Here we want to continue support the legacy property. So the condition should consider both legacy and new for the time being. Similar to the `ConditionalOnSecondaryStorage`

Alternately, if it can use the `OperateProperties.getDatabase()` here instead of the referring to the property key directly, that would be ideal.